src/collection.ts

```diff
@@ -801,30 +801,30 @@ export default class Collection {
       throw new Error("Invalid argument, expected a positive integer.");
     }
     // Retrieve history and check it covers the required time range.
-    const changes = await this.listChangesBackTo<T>(at);
+    const changes = await this.listChangesUntil<T>(at);
     // Replay changes to compute the requested snapshot.
-    const seenIds = new Set();
-    let snapshot: T[] = [];
+    const recordsById = new Map();
     for (const {
       action,
       target: { data: record },
     } of changes) {
       if (action == "delete") {
-        seenIds.add(record.id); // ensure not reprocessing deleted entries
-        snapshot = snapshot.filter((r) => r.id !== record.id);
-      } else if (!seenIds.has(record.id)) {
-        seenIds.add(record.id);
-        snapshot.push(record);
+        recordsById.delete(record.id);
+      } else {
+        recordsById.set(record.id, record);
       }
     }
+    console.debug(`Found ${recordsById.size} records in snapshot (at=${at}).`);
```

Comment:
Should we keep this?

Reply:
No, I agree, this shouldn't be the first one